web-client/scripts/build.sh

```diff
@@ -29,47 +51,68 @@ function generate() {
     BINDGEN_OUTPUT="$OUT_DIR/index_bg.wasm"
     OPT_OUTPUT="$BINDGEN_OUTPUT.opt"
 
-    printf "Generating $TYPE WASM for target $TARGET...\n"
+    echo "Generating $TYPE WASM for target $TARGET...\n"
 
     # Generate JS bindings (also generates a GC'd WASM file)
     wasm-bindgen --weak-refs --target $TARGET --out-name index --out-dir $OUT_DIR --no-typescript $CARGO_OUTPUT
-    # Optimize bindgen's WASM output for size
-    wasm-opt -Os -o $OPT_OUTPUT $BINDGEN_OUTPUT
-    # Replace bindgen's WASM output with the optimized one
-    mv $OPT_OUTPUT $BINDGEN_OUTPUT
+
+    if $RUN_WASM_OPT ; then
+        # Optimize bindgen's WASM output for size
+        wasm-opt -Os -o $OPT_OUTPUT $BINDGEN_OUTPUT
+        # Replace bindgen's WASM output with the optimized one
+        mv $OPT_OUTPUT $BINDGEN_OUTPUT
+    fi
 }
 
 # Prepare build environment
 if ! command -v wasm-bindgen &> /dev/null
 then
-    printf "Installing wasm-bindgen...\n"
+    echo "Installing wasm-bindgen...\n"
     cargo install wasm-bindgen-cli
 fi
 if ! command -v wasm-opt &> /dev/null
 then
-    printf "Installing wasm-opt...\n"
+    echo "Installing wasm-opt...\n"
     cargo install wasm-opt
 fi
 
+contains() { case $2 in *$1* ) return 0;; *) return 1;; esac ;}
+
 # Primitives
 compile "primitives"
-generate "main" "bundler"
-generate "main" "web"
-generate "main" "nodejs"
+if contains "bundler" "$TARGETS"; then
+    generate "main" "bundler"
+fi
+if contains "web" "$TARGETS"; then
+    generate "main" "web"
+fi
+if contains "nodejs" "$TARGETS"; then
+    generate "main" "nodejs"
+fi
 
 # Client
 compile "client"
-generate "worker" "bundler"
-generate "worker" "no-modules" "./dist/web/worker-wasm"
-generate "worker" "nodejs"
+if contains "bundler" "$TARGETS"; then
+    generate "worker" "bundler"
+fi
+if contains "web" "$TARGETS"; then
+    generate "worker" "no-modules" "./dist/web/worker-wasm"
+fi
+if contains "nodejs" "$TARGETS"; then
+    generate "worker" "nodejs"
+fi
 
 # Types
-printf "Building types...\n"
-compile "client,primitives"
-wasm-bindgen --weak-refs --target web --out-name web --out-dir dist/types $CARGO_OUTPUT
-wasm-bindgen --weak-refs --target bundler --out-name bundler --out-dir dist/types $CARGO_OUTPUT
-find dist/types ! -name 'web.d.ts' ! -name 'bundler.d.ts' -type f -exec rm {} +
+if $BUILD_TYPES ; then
```

Comment:
This runs the command in `$BUILD_TYPES`, I'd consider that unexpected. Could you change the variable to contain 0 or 1 instead and compare with `[[` or `[`?

Reply:
This was actually changed by comment from JD: https://github.com/nimiq/core-rs-albatross/pull/2070#discussion_r1434679800

I'd also prefer my original version with string comparison.

Reply:
`true` and `false` are valid commands. I'd say this is expected and the string comparison is not needed

Reply:
The current statement works.

I found it unexpected that the command inside a variable is run, instead of compared as string. If I need to assess the security of this, I need to find out all places where the variable is set, to ensure that nothing unexpected can happen. If it's a string comparison, I don't even need to look.

Reply:
Mmm, I see your point but in the script there are only two places where we're setting this variable and they are only set to `true` and `false`. Having said this, I don't have a strong opinion on this so we could go to the previous version.